hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/AsyncServerSocketTest.java

```diff
@@ -151,49 +152,46 @@ public void test_bind_whenAlreadyBound() {
         socket.close();
     }
 
-//    @Test
-//    public void test_accept_whenConsumerNull() {
-//        Reactor reactor = newReactor();
-//        AsyncServerSocket socket = reactor.newAsyncServerSocketBuilder().build();
-//
-//        SocketAddress local = new InetSocketAddress("127.0.0.1", 5000);
-//        socket.bind(local);
-//
-//        assertThrows(NullPointerException.class, () -> socket.start(null));
-//
-//        socket.close();
-//    }
-//
-//    @Test
-//    public void test_createCloseLoop_withSamereactor() {
-//        SocketAddress local = new InetSocketAddress("127.0.0.1", 5000);
-//        Reactor reactor = newReactor();
-//        for (int k = 0; k < 1000; k++) {
-//            System.out.println("at:" + k);
-//            AsyncServerSocket socket = reactor.newAsyncServerSocketBuilder()
-//                    .set(AsyncSocketOptions.SO_REUSEPORT,true)
-//                    .build();
-//            socket.bind(local);
-//            socket.start(socket1 -> {
-//            });
-//            socket.close();
-//        }
-//    }
-//
-//    @Test
-//    public void test_createCloseLoop_withNewreactor() {
-//        SocketAddress local = new InetSocketAddress("127.0.0.1", 5000);
-//        for (int k = 0; k < 1000; k++) {
-//            System.out.println("at:" + k);
-//            Reactor reactor = newReactor();
-//            reactors.remove(reactor);
-//            AsyncServerSocket socket = reactor.newAsyncServerSocketBuilder()
-//                    .set(AsyncSocketOptions.SO_REUSEPORT,true)
-//                    .build();
-//            socket.bind(local);
-//            socket.start(socket1 -> {
-//            });
-//            terminate(reactor);
-//        }
-//    }
+    @Test
+    public void test_createCloseLoop_withSamereactor() {
+        SocketAddress local = new InetSocketAddress("127.0.0.1", 5000);
+        Reactor reactor = newReactor();
+        for (int k = 0; k < 1000; k++) {
+            System.out.println("at:" + k);
+            AsyncServerSocket serverSocket = reactor.newAsyncServerSocketBuilder()
+                    .setAcceptConsumer(acceptRequest -> {
+                        AsyncSocket clientSocket = reactor.newAsyncSocketBuilder(acceptRequest)
+                                .setReadHandler(new DevNullReadHandler())
+                                .build();
+                        clientSocket.start();
+                    })
+                    .set(AsyncSocketOptions.SO_REUSEPORT, true)
+                    .build();
+            serverSocket.bind(local);
+            serverSocket.start();
+            serverSocket.close();
+        }
+    }
+
+    @Test
+    public void test_createCloseLoop_withNewReactor() {
+        SocketAddress local = new InetSocketAddress("127.0.0.1", 5000);
+        for (int k = 0; k < 1000; k++) {
+            System.out.println("at:" + k);
+            Reactor reactor = newReactor();
+            reactors.remove(reactor);
```

Comment:
Do we need to remove the created reactor from this list? I mean, as far as I can tell, it is not a problem to call terminate on an already terminated reactor. But, if there is an exception thrown before the loop iteration is done, we will not clean up the reactor created in that iteration of the loop

Reply:
The reactors can consume quite a lot of memory. So removing it from the reactor list is a "better safe than sorry action".

Reply:
Then, can we move it to the end of the loop iteration? As I said, if any of the lines below would throw, (like the `            serverSocket.bind(local);` line), the reactor won't be terminated, because neither we call `terminate(reactor)` nor the After method (because the reactor is removed from the list)

Reply:
Good idea. Fixed.